lib/services/credentials/credential.dao.js

```diff
@@ -145,20 +145,47 @@ dao.removeCredential = function (id, type) {
   return db.del(buildIdKey(type, id));
 };
 
+/*
+ * Remove all credentials
+ *
+ * @id {String}
+ */
 dao.removeAllCredentials = function (id) {
   const dbTransaction = db.multi();
   const credentialTypes = Object.keys(config.models.credentials.properties);
-  const awaitAllPromises = [];
-  credentialTypes.forEach((type) => {
+  const awaitAllPromises = credentialTypes.map(async type => {
+    const authKey = buildIdKey(type, id);
+
     if (type === 'key-auth') {
       // id in this call is actually consumerId, so we need to get all referenced keyIds
-      awaitAllPromises.push(db.smembers(buildIdKey(type, id)).then(ids => {
-        ids.map(keyId => {
-          dbTransaction.del(buildIdKey(type, keyId));
-        });
-      }));
+      // This returns a list of all keys the user owns
+      const ids = await db.smembers(authKey);
+
+      // Get all the scope so we can remove key from them
+      const scopes = await dao.getAllScopes();
+
+      // Delete each key and remove key from scopes if they exist
+      const promises = [];
+      ids.forEach(keyId => {
```

Comment:
Maybe this can become another `.map`?

Reply:
Map won't work here because we loop through the scopes for each key as well. So it has the potential to return more than one item.

In case you didn't notice, I also fixed the scopes thing since that wasn't being cleaned up on user delete either.